### ISSUE TITLE
使用CPU训练检测过程，速度巨慢，请问是哪里配置出错了吗？

### ISSUE THREAD

**Issue by GanWang** [PaddlePaddle/PaddleOCR#4560]
[2021/11/05 13:25:35] root INFO: train dataloader has 1085 iters
[2021/11/05 13:25:35] root INFO: valid dataloader has 500 iters
[2021/11/05 13:25:35] root INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations
[2021/11/05 13:25:35] root INFO: Initialize indexs of datasets:['E:/TXSJ/image/files_2twoall/out_label.txt']

[2021/11/05 13:37:49] root INFO: epoch: [1/100], iter: 10, lr: 0.000100, loss: 7.172621, loss_shrink_maps: 4.851520, loss_threshold_maps: 1.352844, loss_binary_maps: 0.965043, reader_cost: 0.07199 s, batch_cost: 73.44047 s, samples: 88, ips: 0.11982

[2021/11/05 13:48:58] root INFO: epoch: [1/100], iter: 20, lr: 0.000100, loss: 6.892204, loss_shrink_maps: 4.807907, loss_threshold_maps: 1.160244, loss_binary_maps: 0.957401, reader_cost: 0.00000 s, batch_cost: 66.81513 s, samples: 80, ips: 0.11973

[2021/11/05 14:00:07] root INFO: epoch: [1/100], iter: 30, lr: 0.000100, loss: 6.733838, loss_shrink_maps: 4.708129, loss_threshold_maps: 1.091049, loss_binary_maps: 0.927331, reader_cost: 0.00020 s, batch_cost: 66.97297 s, samples: 80, ips: 0.11945

config文件det_r50_vd_db里面是：
Global:
  use_gpu: False
  epoch_num: 100
  #epoch_num: 1200
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/det_r50_vd/
  save_epoch_step: 100
  #save_epoch_step: 1200
  # evaluation is run every 2000 iterations after the 0th iteration
  eval_batch_step: [0, 2000]
  cal_metric_during_train: False
  pretrained_model: ./pretrain_models/ResNet50_vd_ssld_pretrained
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./output/det_db/predicts_db.txt

Architecture:
  model_type: det
  algorithm: DB
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: DBFPN
    out_channels: 256
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: True
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.0001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.7
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: E:/TXSJ/image/files_2twoall/imgs/
    label_file_list:
      - E:/TXSJ/image/files_2twoall/out_label.txt
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
            - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
      - EastRandomCropData:
          size: [640, 640]
          max_tries: 50
          keep_ratio: True
      - MakeBorderMap:
          shrink_ratio: 0.4
          thresh_min: 0.3
          thresh_max: 0.7
      - MakeShrinkMap:
          shrink_ratio: 0.4
          min_text_size: 8
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 8
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: E:/TXSJ/icdar2015/detection/test/imgs/
    label_file_list:
      - E:/TXSJ/icdar2015/detection/test/out_label.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
          image_shape: [736, 1280]
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 1

**Comment by GanWang** [comment]
![image](https://user-images.githubusercontent.com/39181015/140466110-088bb269-55cf-4be0-aee6-1e212a068eac.png)

**Comment by GanWang** [comment]
![image](https://user-images.githubusercontent.com/39181015/140466136-60697314-0d0d-497f-9eb2-4987165e986d.png)
CPU的使用率也很低，训练速度也巨慢

**Comment by Gmgge** [comment]
设计cpu使用率应该是相关调度问题，可能需要官方人员解答，通常可以增加batch size来拉升该使用率，但是你这内存已经到瓶颈了。
不过还是有一点小意见，希望可以提供到帮助：
1.不建议使用cpu来训练resnet50骨干模型，建议更换为mobilenet骨干模型，计算复杂度将大大减少。
2.如果想要训练resnet50，可以尝试使用百度的云服务器，或者colab，或者其他拥有GPU的机器。

**Comment by littletomatodonkey** [comment]
可以试着把num_workers 修改为4，之类的，resnet50确实也太大了，训练的话建议使用aistudio资源

**Comment by GanWang** [comment]
> 可以把num_workers修改为4，之类的，resnet50确实也关注了，训练的话建议使用aistudio资源

num_workers是数据加载的线程数量？
